hardware_interface/include/hardware_interface/joint_state_interface.h

```diff
@@ -64,18 +140,75 @@ class JointStateHandle
     {
       throw HardwareInterfaceException("Cannot create handle '" + name + "'. Effort data pointer is null.");
     }
+
   }
 
   std::string getName() const {return name_;}
   double getPosition()  const {assert(pos_); return *pos_;}
   double getVelocity()  const {assert(vel_); return *vel_;}
   double getEffort()    const {assert(eff_); return *eff_;}
 
+  const double* getPositionPtr() const {return pos_;}
+  const double* getVelocityPtr() const {return vel_;}
+  const double* getEffortPtr()   const {return eff_;}
+
+  double getAbsolutePosition() const {
+    assert(absolute_pos_);
+    if(!hasAbsolutePosition()){
+      throw std::runtime_error("Joint state handle does not contain absolute encoder position information");
+    }
+    return *absolute_pos_;
+  }
+
+  const double* getAbsolutePositionPtr() const{
+    assert(hasAbsolutePosition());
+    if(!hasAbsolutePosition()){
+      throw std::runtime_error("Joint state handle does not contain torque sensor information");
+    }
+    return absolute_pos_;
+  }
+
+  double getTorqueSensor() const {
+    assert(torque_sensor_);
+    if(!hasTorqueSensor()){
+      throw std::runtime_error("Joint state handle does not contain torque sensor information");
+    }
+    return *torque_sensor_;
+  }
+
+  const double* getTorqueSensorPtr() const {
+    assert(torque_sensor_);
+    if(!hasTorqueSensor()){
+      throw std::runtime_error("Joint state handle does not contain torque sensor information");
+    }
+    return torque_sensor_;
+  }
+
+  bool hasAbsolutePosition() const {
+    if(absolute_pos_){
+      return true;
+    }
+    return false;
+  }
```

Comment:
As in `actuator_state_interface`, simplify these two funcs to single-line `return absolute_pos_;`/`return torque_sensor_;`.

Reply:
I guess this was artistic freedom ;)